plugin/src/main/java/org/wildfly/plugins/bootablejar/maven/cli/ForkedCLIUtil.java

```diff
@@ -59,8 +59,15 @@ public static void fork(Log log, String[] artifacts, Class<?> clazz, Path home,
         for (String loc : artifacts) {
             cp.append(loc).append(File.pathSeparator);
         }
-        collectCpUrls(getJavaHome(), Thread.currentThread().getContextClassLoader(), cp);
-
+        StringBuilder contextCP = new StringBuilder();
+        collectCpUrls(getJavaHome(), Thread.currentThread().getContextClassLoader(), contextCP);
+        // This happens when running tests, use the process classpath to retrieve the CLIForkedExecutor main class
+        if (contextCP.toString().isEmpty()) {
```

Comment:
This is minor really, but this should probably use `contextCp.length() == 0)`. The `toString()` will convert it to a string each time which isn't really needed.

Reply:
Thank-you, yes. fixed.